.github/workflows/build.yml

```diff
@@ -0,0 +1,24 @@
+name: build
+
+on:
+  pull_request:
+      types: [ opened, synchronize, reopened, ready_for_review, labeled, unlabeled ]
+  push:
+    branches: [ master, 113-setup-build-and-test-with-github-actions]
+    
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    container: rust:1.53.0
+    env:
+      RUST_TC_VER: nightly-2021-06-23
+    steps:
+      - name: Checks-out repository 
+        uses: actions/checkout@v2
+      - name: Build
+        run: |
+          export PEARL_COMMIT_HASH=$(echo ${GITHUB_SHA} | cut -c1-8)
+          rustup install ${{ env.RUST_TC_VER }}
+          rustup default ${{ env.RUST_TC_VER }}
+          cargo build --all
```

Comment:
We can try these actions here: 
https://github.com/marketplace/actions/rust-cargo
https://github.com/marketplace/actions/rust-toolchain